src/main/java/me/oondanomala/fpkmod/movement/ParkourHandler.java

```diff
@@ -124,5 +128,21 @@ static void update(EntityPlayerSP player, PlayerState pastState, PlayerState sec
         if (secondPastState.onGround && !pastState.onGround && player.movementInput.moveStrafe != 0 && !pastState.isStrafing()) {
             last45 = player.rotationYaw - pastState.yaw;
         }
+
+        // Sidestep
+        if (isJumpTick) {
+            boolean strafingLeft = player.movementInput.moveStrafe > 0.0;
+            if (pastState.isStrafing() && (strafingLeft != pastState.keyLeft)) {
```

Comment:
```suggestion
            if (pastState.isStrafing() && (player.movementInput.moveStrafe == 0)) {
```
I *think* that fixes it, but i haven't tested it much

Reply:
The error is mostly due to an oversight on my part. I forgot to test the left side and realized that i just forgot to check if the player was strafing at all on the current tick. I addressed this issue and the wawd vs wdwa naming in my latest commit to this branch. Also I will make sure to make feature branches in the feature, it is a pleasure working with you on this project :D. And about the "jankiness" it might be due to the fact that the sidestep is updated in real time unlike CyvForge's sidestep label, which makes it so that when you jump even if you don't strafe it will still give you a wad sidestep for the amount of ticks you were in the air. I can change this if you would like.

Reply:
Is the old `strafingLeft` check still needed?

Aside from that i think the logic is fine for now, the jank can be fixed later when last timing gets implemented (since i'd expect some overlap in the code)

Reply:
It is, the strafingLeft check is just to check if the player is strafing in a different direction than the one they were strafing in before the jump. I started to work on the last timing and there will be some overlap between last timing and input history, since they are using the same list of inputs. However, I don't believe there is going to be overlap between last timing and last sidestep.

Reply:
It's fine then